zk.js/src/account.ts

```diff
@@ -425,46 +389,27 @@ export class Account {
 
   static getEncryptionKeyPair(seed: string): nacl.BoxKeyPair {
     const encSeed = seed + "encryption";
-    const encryptionPrivateKey = blake2b
-      .create(b2params)
-      .update(encSeed)
-      .digest();
+    const encryptionPrivateKey = blake(encSeed, Account.hashLength);
     return nacl.box.keyPair.fromSecretKey(encryptionPrivateKey);
   }
 
-  static generateShieldedPrivateKey(seed: string, poseidon: any): BN {
-    const privkeySeed = seed + "shielded";
-    const privateKey = new BN(
-      poseidon.F.toString(
-        poseidon([
-          new BN(blake2b.create(b2params).update(privkeySeed).digest()),
-        ]),
-      ),
-    );
-    return privateKey;
+  static generateShieldedPrivateKey(seed: string, poseidon: Poseidon): BN {
+    const privateKeySeed = seed + "shielded";
+    const blakeHash: Uint8Array = blake(privateKeySeed, Account.hashLength);
+    const blakeHash31 = truncateToCircuit(blakeHash);
```

Comment:
you can unify lines 398 and 399 by just setting the blake hash length to 31 bytes.
Below is just an example using a constant is better ofc.
```suggestion
    const blakeHash: Uint8Array = blake(privateKeySeed,31);
```